src/features/completions.zig

```diff
@@ -1416,7 +1416,8 @@ fn completeFileSystemStringLiteral(
 }
 
 pub fn completionAtIndex(server: *Server, analyser: *Analyser, arena: std.mem.Allocator, handle: *const DocumentStore.Handle, source_index: usize) error{OutOfMemory}!?types.CompletionList {
-    const at_line_start = offsets.lineSliceUntilIndex(handle.tree.source, source_index).len == 0;
+    const line_slice = offsets.lineSliceUntilIndex(handle.tree.source, source_index);
+    const at_line_start = line_slice.len == 0 or line_slice.len == 1;
```

Comment:
do snippet completions not appear for you without this change? I just checked VSCode, ST4 and Helix and it worked fine before.
```suggestion
    const at_line_start = offsets.lineSliceUntilIndex(handle.tree.source, source_index) <= 1;
```

Reply:
I even get completions without explicitly requesting them:

just typing 'main':
![A](https://github.com/zigtools/zls/assets/19954306/bd3306d1-c23b-4182-b360-f4ebe906892a)
explicitly request completions then type 'main':
![B](https://github.com/zigtools/zls/assets/19954306/940b6a52-d3aa-40f2-b50b-f508f79d66e8)
type 'm' the request completions:
![C](https://github.com/zigtools/zls/assets/19954306/0aca9f96-cf8d-4551-80bc-251bdad55249)